ocfweb/account/templates/account/register/index.html

```diff
@@ -118,9 +123,76 @@
                 </p>
             </div>
 
-            <hr />
+            <div class="well well-sm">
+                <strong>Available Account Names</strong>
+                <ul class="list-group list-group-flush" id="recommendations" style="margin: 7px"></ul>
+                <button type="button" class="btn btn-secondary" onclick="recommend()">More</button>
+            </div>
+            <hr class="visible-sm" />
 
             {% include "account/partials/password-requirements-sidebar.html" %}
+
         </div>
     </div>
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
+    <script>
+        $(document).ready(function() {
+            var finTypingCountdown = 250; // 250 milliseconds
+            var typingTimer;
+            var $input = $('#id_ocf_login_name');
+
+            // On keyup, start countdown
+            $input.keyup(function() {
+                //alert('key UP');
+                clearTimeout(typingTimer);
+                typingTimer = setTimeout(validate_username, finTypingCountdown);
+            });
+
+            // On keydown, clear countdown
+            $input.keydown(function() {
+                //alert('key DOWN');
+                clearTimeout(typingTimer);
+            });
+
+            function validate_username() {
+                $.ajax({
+                     type: 'GET',
+                     url: 'validate',
+                     data: {'username': $input.val(),
+                            'real_name': '{{real_name}}'},
+                     success: function(data) {
+                         if(data.is_valid) {
+                             $input.css('border', '5px solid #3c763d');
+                             $('#valid_username_notif').css('display', 'block')
+                             $('#invalid_username_notif').css('display', 'none')
+                         } else {
+                             $input.css('border', '5px solid #a94442');
+                             $('#invalid_username_notif').css('display', 'block')
+                             $('#valid_username_notif').css('display', 'none')
+                         }
+                     }
+                 });
+             }
+        });
+    </script>
+    <script>
+        function recommend() {
+            $.ajax({
+                 type: 'GET',
+                 url: 'recommend',
+                 data: {'real_name': '{{real_name}}'},
+                 success: function(data) {
+                     $('#recommendations').empty();
+                     for (var i in data) {
+                         $('#recommendations').append(
+                             '<li class="list-group-item list-group-item-success">'+ data[i] +'</li>'
+                         );
+                     }
+                 }
+             });
+        }
+        $(document).ready(function() {
+            recommend();
+        });
+    </script>
```

Comment:
Can we try to move most/all of this under [`ocfweb/static/js/pages`](https://github.com/ocf/ocfweb/tree/master/ocfweb/static/js/pages/) instead of making it an inline script? Perhaps we could inline only a stub script that sets the `real_name` variable.